java/client/src/main/java/glide/utils/ArrayTransformUtils.java

```diff
@@ -34,4 +48,16 @@ public static <T, U extends T> U[] castArray(T[] objectArr, Class<U> clazz) {
                 .map(clazz::cast)
                 .toArray(size -> (U[]) Array.newInstance(clazz, size));
     }
+
+    /**
+     * Concatenates multiple arrays of type T and returns a single concatenated array.
+     *
+     * @param arrays Varargs parameter for arrays to be concatenated.
+     * @param <T> The type of the elements in the arrays.
+     * @return A concatenated array of type T.
+     */
+    @SafeVarargs
+    public static <T> T[] concatenateArrays(T[]... arrays) {
+        return Stream.of(arrays).flatMap(Stream::of).toArray(size -> Arrays.copyOf(arrays[0], size));
```

Comment:
Can you try this?
```suggestion
        return Stream.of(arrays).flatMap(Stream::of).toArray(T[]::new);
```

Reply:
It does not allow for generic array creation this way.